src/jit.h

```diff
@@ -36,6 +36,11 @@ struct jit_state {
     int n_jumps;
 };
 
+struct host_reg {
+    int reg_name; /* enum x64_reg/a64_reg */
```

Comment:
How about using `uint8_t` for `reg_name`?

Reply:
This variable is indeed an index rather than a name for registers.

Reply:
> This variable is indeed an index rather than a name for registers.

In fact, it is the index to the host's register file, but not the index of the `register_map` array. i.e. the third register (index: 2, zero-based) of x64 is `RCX`, but the one in `register_map` is `RDX`. We want to retrieve the `dirty` flag through the register name `RDX` instead of its index to `register_map`.

Reply:
> By shifting the value left by 1, we can incorporate both the register index value and its 'dirty' status within a short integer. Consequently, checking the 'dirty' status becomes a straightforward bitwise AND operation with `0x1`.

Can we define a new `struct` here in C? Use `int8_t` might be too confusing.

Reply:
> Can we define a new `struct` here in C? Use `int8_t` might be too confusing.

Yes, use it with bit field.